diag_input.c

```diff
@@ -242,30 +235,32 @@ struct radio_message * handle_4G(struct diag_packet *dp, unsigned len)
 			break;
 		case 7: // UL-CCCH
 			m->chan_nr = 2;
+			m->bb.arfcn[0] |= ARFCN_UPLINK;
 			break;
 		case 8: // UL-DCCH
 			m->chan_nr = 3;
+			m->bb.arfcn[0] |= ARFCN_UPLINK;
 			break;
 		default:
 			// Unhandled
 			return NULL;
 		}
 		// verify len
-		payload_len = ((uint16_t)dp->data[9]) << 8 | dp->data[8];
-		if (payload_len > len - 15) {
+		payload_len = dp->data[12]-1;
```

Comment:
Are you sure that the length field is only 8bit? I would expect 16bit here?

Reply:
This length does not reflect the actual size of the message anyway.
This is corrected in a subsequent commit

Reply:
But these commits are not in this pull request yet? I am not a LTE expert but I would expect QXDM to have a 16bit length field and LTE to have longer messages too. bb.data can be > 256 as well.

I can merge anyway but I would expect us to see truncated messages?

Reply:
There are 4 commits in this pull request
https://github.com/moiji-mobile/diag-parser/pull/6/commits

Reply:
Yes. but I look at files changed which is "git diff base..4thcommit"? So payload_len = dp->data[12]-1 is the final one. Do you have an indication of why this field only needs to be 8bit and not 16bit?

Reply:
with all 4 commits, it looks like this :
		/* payload_len = dp->data[12]-1; */
		payload_len=len-14;
		printf("payload_len: %d\n",payload_len);
		data = &dp->data[14];

Reply:
I should probably remove the comment

Reply:
Okay. Not sure how I ended up on the patch. I thought I clicked PR->files changed.. Hmm. Could you remove the /* payload_len .. */ and then merge it? Thank you!